src/scraper/__main__.py

```diff
@@ -1,35 +1,81 @@
 import logging, os, re, csv, urllib.parse, json, sys
-from .arguments import args
 from datetime import datetime, timedelta
 from time import time
 import requests
 from bs4 import BeautifulSoup
 from .helpers import *
 
 class scraper:
-
-    def __init__(self, county, start_date, end_date):
+    def __init__(self, # The parameters are the defaults.
+                 county = 'hays',
+                 start_date = '2024-07-01', 
+                 end_date = '2024-07-01', 
+                 case_number = None, 
+                 judicial_officers = [],
+                 ms_wait = 200,
+                 no_overwrite = False,
+                 log = "INFO",
+                 court_calendar_link_text = "Court Calendar",
+                 test = False,
+                 location = False
+                 ):
+        
+        # These are optional input fields, with a bit of validation.
         self.county = county.lower()
         self.start_date = datetime.strptime(start_date, '%Y-%m-%d').date()
         self.end_date = datetime.strptime(end_date, '%Y-%m-%d').date()
-        self.session = requests.Session()
+        self.ms_wait = ms_wait #"Number of ms to wait between requests."
+        self.judicial_officers = judicial_officers #"Judicial Officers to scrape. For example, -j 'mr. something' 'Rob, Albert'. By default, it will scrape all JOs.",
+        self.no_overwrite = no_overwrite #"Switch to don't overwrite cached html files if you want to speed up the process (but may not get the most up to date version).",
+        self.log = log #"Set the level to log at.",
+        self.court_calendar_link_text = court_calendar_link_text #"This is the link to the Court Calendar search page at default.aspx, usually it will be 'Court Calendar', but some sites have multiple calendars e.g. Williamson",
+        self.test = test #"If this parameter is present, the script will stop after the first case is scraped.",
+        self.case_number = case_number #"If a case number is entered, only that single case is scraped. ex. 12-2521CR",
+        self.location = location # ??
+
+        # These are class scraper-level fields that need to be stored and used from function to function. 
+        self.odyssey_version = None
+        self.base_url = None
+        self.main_soup = None
+        self.main_page_html = None
+        self.search_url = None
+        self.search_page_html = None
+        self.search_soup = None
+        self.hidden_values = None
+        self.judicial_officer_to_ID = None
```

Comment:
Might be helpful to add type annotations to these values (and the other instance values above).

Reply:
Will do!